students/timm67/session04/trigrams.py

```diff
@@ -1,22 +1,179 @@
 #!/usr/bin/env python3
 
-words = "I wish I may I wish I might".split()
+import random
+import sys
 
-def build_trigrams(words):
-    """
-    build up the trigrams dict from the list of words
+trigrams = {}
 
-    returns a dict with:
-       keys: word pairs
-       values: list of followers
-    """
-    trigrams = {}
 
-    # build up the dict here!
+def build_trigrams(word_list, trigrams_dict):
+    if word_list is None:
+        print('build_trigrams: No words to build trigram dict')
+        return
 
-    return trigrams
+    if (len(word_list) < 3):
+        print('build_trigrams: Not enough words to build trigram')
+        return
+
+    for i in range(0, len(word_list)-2, 1):
+        key = (word_list[i], word_list[i+1])
+        if key in trigrams_dict.keys():
+            trigrams_dict[key].append(word_list[i+2])
+        else:
+            trigrams_dict[key] = [word_list[i+2]]
+
+
+def build_trigrams_list(words_list, trigrams_dict):
+    for line_list in words_list:
+        build_trigrams(line_list, trigrams_dict)
```

Comment:
Minor: This function doesn't seem to be called from anywhere, so it better be removed.